### PR TITLE
Add CMAKE_ARGS to docs

### PR DESCRIPTION
**Pull request by BastianZim**
PR Checklist:

- [x] make all edits to the docs in the `src` directory, not in `docs` or in the html files
- [x] note any issues closed by this PR with [closing keywords](https://help.github.com/articles/closing-issues-using-keywords)
- [x] put any other relevant information below

CC @chrisburr
xref https://github.com/conda-forge/libxc-feedstock/pull/27#issuecomment-966192705

### CODE REVIEW

File: src/maintainer/knowledge_base.rst [hunk 1]
```diff
@@ -45,6 +45,7 @@ Some optional, but useful CMake options:
     - ``-DCMAKE_INSTALL_LIBDIR=lib`` Libraries will land in $PREFIX/lib, sometimes projects install
       into lib64 or similar but on conda-forge we keep shared libraries in simply lib.
     - ``-DBUILD_SHARED_LIBS=ON`` Instruct CMake to build shared libraries instead of static ones.
+    - ``${CMAKE_ARGS}`` Add variables defined by conda-forge internally. This is required to enable various conda-forge enhancements, like `CUDA builds <https://conda-forge.org/docs/maintainer/knowledge_base.html#cuda-builds>`__.
```
Comment (BastianZim) [inline src/maintainer/knowledge_base.rst:48]: Not completely sure about the wording as I don't fully know what `CMAKE_ARGS` does. Happy about any suggestions.

**Comment by BastianZim** [comment]
@conda-forge/core Quick ping here.

**Review by jakirkham — APPROVED** [review]

File: src/maintainer/knowledge_base.rst (hunk 1 above)
Comment (jakirkham) [reply src/maintainer/knowledge_base.rst:48]: This seems fine. The main thing is we have some text in here. People can always iterate and refine this later

**Comment by jakirkham** [comment]
Thanks Bastian! 😄

**Comment by BastianZim** [comment]
Thanks for merging John!

**Pull request merged**